src/Workleap.Extensions.Mongo/Indexing/MongoIndexer.cs

```diff
@@ -94,6 +96,11 @@ private async Task UpdateIndexesInternalAsync(IEnumerable<Type> types, IMongoDat
     {
         var registry = new IndexRegistry(types);
```

Comment:
I find a bit weird that for the Attributes approached it's the IndexRegistry that is responsible to extract the Indexer from a Type but for the ClassMap it is done by the MongoIndexer. 

I think it would cleaner to move the logic out of the IndexRegistry (maybe here or an other static method that can be easilty tested), and only call registry.AddDocumentTypeEntry() that way it is unified.